File: 2022-2023/sourcecode/lab/lab3/RecyclerViewSelectorJava/app/src/main/java/pl/edu/uwr/pum/recyclerviewselectorjava/NumberItemDetailsLookup.java
package pl.edu.uwr.pum.recyclerviewselectorjava;

import android.view.MotionEvent;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.selection.ItemDetailsLookup;
import androidx.recyclerview.widget.RecyclerView;

public class NumberItemDetailsLookup extends ItemDetailsLookup<Long> {

    private final RecyclerView recyclerView;

    public NumberItemDetailsLookup(RecyclerView recyclerView) {
        this.recyclerView = recyclerView;
    }

    @Nullable
    @Override
    public ItemDetails<Long> getItemDetails(@NonNull MotionEvent e) {
        View view = recyclerView.findChildViewUnder(e.getX(), e.getY());
        if (view != null)
            return ((NumberListAdapter.NumberListViewHolder)recyclerView.getChildViewHolder(view)).getItemDetails();
        return null;
    }
}

File: 2022-2023/sourcecode/lab/lab3/RecyclerViewSelectorKotlin/app/src/main/java/pl/edu/uwr/pum/recyclerviewselectorkotlin/NumberItemDetailsLookup.kt
package pl.edu.uwr.pum.recyclerviewselectorkotlin

import android.view.MotionEvent
import androidx.recyclerview.selection.ItemDetailsLookup
import androidx.recyclerview.widget.RecyclerView

class NumberItemDetailsLookup(private val recyclerView: RecyclerView) :
    ItemDetailsLookup<Long>() {
    override fun getItemDetails(event: MotionEvent): ItemDetails<Long>? {
        val view = recyclerView.findChildViewUnder(event.x, event.y)
        if (view != null) {
            return (recyclerView.getChildViewHolder(view) as NumberListAdapter.NumberListViewHolder)
                .getItemDetails()
        }
        return null
    }
}